tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/TableDefinitionsRepository.java

```diff
@@ -99,14 +101,73 @@ public TableDefinitionsRepository(Collection<TableDefinition> tableDefinitions)
 
     public <T extends TableDefinition> T register(T tableDefinition)
     {
-        checkState(!tableDefinitions.containsKey(tableDefinition.getName()), "duplicated table definition: %s", tableDefinition.getName());
-        tableDefinitions.put(tableDefinition.getName(), tableDefinition);
+        TableDefinitionRepositoryKey repositoryKey = asRepositoryKey(tableDefinition.getTableHandle());
+        checkState(!tableDefinitions.containsKey(repositoryKey), "duplicated table definition: %s", repositoryKey);
+        tableDefinitions.put(repositoryKey, tableDefinition);
         return tableDefinition;
     }
 
-    public TableDefinition get(String name)
+    public TableDefinition get(TableHandle tableHandle)
```

Comment:
I prefer not to add any new concept like `uniqueName`. I would better stick with data base concepts like table name and schema. 

I don't want to have tests to be as much explicit as possible, because in case of TPCH and TPCDS I want to have queries as less modified as possible, unmodified is the goal. As I understand I would need to have an unique name for customer table, and then I would need to modify query to access that table by unique name. 

Lets talk about this offline. 

I want to close this issue and do the release to continue the work on TPCDS product tests for presto, lets continue the work on this as separate pull request.